### PR TITLE
Support for v1 and v2

### PR DESCRIPTION
**Pull request by kushal256**
Previous has support for v1, now added v2 so both are supported simultaneously.

### CODE REVIEW

File: bitgo-java-api/build.gradle [hunk 1]
```diff
@@ -1,14 +1,17 @@
 apply plugin:'java'
 apply plugin:'idea'
-version='0.0.4'
+
+group = 'com.bitso'
+version='SNAPSHOT'
```
Comment (trailtacos) [inline bitgo-java-api/build.gradle:5]: This doesn't seem right

File: bitgo-java-api/build.gradle [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+    maven { url 'https://jitpack.io' }
 }
 
 dependencies {
-    compileOnly 'org.projectlombok:lombok:1.16.16'
-    compile 'org.slf4j:slf4j-api:1.7.25',
-            'com.alibaba:fastjson:1.2.31'
+    compile 'org.projectlombok:lombok:1.+'
+    compile 'org.slf4j:slf4j-api:1.7.+',
+            'com.fasterxml.jackson.core:jackson-databind:2.+'
```
Comment (trailtacos) [inline bitgo-java-api/build.gradle]: You should fix the version (even if they need updating)

File: bitgo-java-api/src/main/java/com/bitso/bitgo/v2/BitGoClientImpl.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    private static final String GET_WALLET_URL = "/$COIN/wallet/";
+    private static final String CURRENT_USER_PROFILE_URL = "/user/me";
+    private static final String GET_WALLET_TRANSFER_URL = "/$COIN/wallet/$WALLET/transfer/$TRANSFER";
+    private static final String GET_WALLET_TRANSFER_SEQ_URL = "/$COIN/wallet/$WALLET/transfer/sequenceId/$SEQUENCE";
+    private static final String LIST_WALLET_TRANSFER_URL = "/$COIN/wallet/$WALLET/transfer";
+    private static final String UNLOCK_URL = "/user/unlock";
+
+    private static final String V1_LIST_WALLET_TXN_URL = "/wallet/$WALLET/tx";
+    /**
+     * The base URL (host, port, up to api/v1 for example
```
Comment (trailtacos) [inline bitgo-java-api/src/main/java/com/bitso/bitgo/v2/BitGoClientImpl.java]: This seems like a wrong comment - this appears to be a v2 API

File: bitgo-java-api/src/main/java/com/bitso/bitgo/v1/BitGoClientImpl.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+     * -H "Content-Type: application/json" \
+     * -H "Authorization: Bearer $ACCESS_TOKEN" \
+     * https://test.bitgo.com/api/v1/wallet/$WALLET/tx
+     *
+     * @param walletId
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public WalletTransactionResponse listWalletTransctions(String walletId, long skip, int limit) throws IOException {
```
Comment (trailtacos) [inline bitgo-java-api/src/main/java/com/bitso/bitgo/v1/BitGoClientImpl.java]: `listWalletTransactions`

File: bitgo-java-api/setting.gradle [hunk 5]
```diff
@@ -0,0 +1,4 @@
+rootProject.name = bitgo
+
+// For publishing ensure we will work with gradle 5.0 https://docs.gradle.org/4.9/userguide/publishing_maven.html#publishing_maven:deferred_configuration
+//enableFeaturePreview('STABLE_PUBLISHING')
```
Comment (trailtacos) [inline bitgo-java-api/setting.gradle]: Does this need to be done?

**Review by trailtacos — CHANGES_REQUESTED** [review]

File: bitgo-java-api/build.gradle (hunk 1 above)
Comment (kushal256) [reply bitgo-java-api/build.gradle:5]: Jitpack will determine version number via the github release tag, not from what we define in our build.gradle file.  This was changed here avoid any potential confusion.

File: bitgo-java-api/build.gradle (hunk 2 above)
Comment (kushal256) [reply bitgo-java-api/build.gradle]: Done, checked in

File: bitgo-java-api/src/main/java/com/bitso/bitgo/v2/BitGoClientImpl.java (hunk 3 above)
Comment (kushal256) [reply bitgo-java-api/src/main/java/com/bitso/bitgo/v2/BitGoClientImpl.java]: Done, checked in

File: bitgo-java-api/src/main/java/com/bitso/bitgo/v1/BitGoClientImpl.java (hunk 4 above)
Comment (kushal256) [reply bitgo-java-api/src/main/java/com/bitso/bitgo/v1/BitGoClientImpl.java]: Done, checked in

File: bitgo-java-api/setting.gradle (hunk 5 above)
Comment (kushal256) [reply bitgo-java-api/setting.gradle]: No, removed this comment.  This was from experimenting with publishing jars, but it's being handled by jitpack now.

File: bitgo-java-api/src/main/java/com/bitso/bitgo/util/HttpHelper.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        }
+        return false;
+    }
+
+    private static void post(HttpURLConnection conn, String auth, Map<String, Object> data)
+            throws IOException {
+
+        conn.setRequestMethod("POST");
+//        conn.setRequestProperty("Content-Type", "application/json");
+//        conn.setRequestProperty("Authorization", "Bearer " + auth);
```
Comment (trailtacos) [inline bitgo-java-api/src/main/java/com/bitso/bitgo/util/HttpHelper.java]: Delete these comments

**Review by trailtacos — APPROVED** [review]
Aside from deleeting the comments, this LGTM

**Pull request merged**